sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AADOAuth2UserService.java

```diff
@@ -115,20 +115,41 @@ public OidcUser loadUser(OidcUserRequest userRequest) throws OAuth2Authenticatio
         if (!properties.allowedGroupIdsConfigured() && !properties.allowedGroupNamesConfigured()) {
             return Collections.emptySet();
         }
-        Set<String> groups = Optional.of(accessToken)
-                                     .map(AbstractOAuth2Token::getTokenValue)
-                                     .map(graphClient::getGroupsFromGraph)
-                                     .orElseGet(Collections::emptySet);
-        if (properties.getUserGroup().getEnableFullList()) {
-            return groups;
+        GroupInformation groupInformation = GroupInformation(accessToken);
+
```

Comment:
How about add local variable here:
```
  List<String> roleStrings = new ArrayList();
```
And put values into it in the following code.